frontend/src/components/common/button/button.tsx

```diff
@@ -1,17 +1,40 @@
+import { AppRoute } from 'common/enums/enums';
 import { FC } from 'common/types/types';
+import { getValidClasses } from 'helpers/helpers';
+import { useNavigate } from 'hooks/hooks';
 
 import styles from './styles.module.scss';
 
 type Props = {
   label: string;
+  btnColor?: 'blue' | 'gray';
   type?: 'button' | 'submit';
+  to?: AppRoute;
   onClick?: () => void;
 };
 
-const Button: FC<Props> = ({ type = 'button', label, onClick }) => (
-  <button type={type} className={styles.button} onClick={onClick}>
-    {label}
-  </button>
-);
+const Button: FC<Props> = ({
+  type = 'button',
+  label,
+  btnColor = 'blue',
+  to,
+  onClick,
+}) => {
+  const navigate = useNavigate();
+
+  const handleOnClick = (): void => {
+    to ? navigate(to) : onClick && onClick();
```

Comment:
```suggestion
    to ? navigate(to) : onClick?.();
```
optional chaining can be used on functions

Reply:
done